builder/tencentcloud/cvm/common.go

```diff
@@ -105,11 +106,20 @@ func GetImageByName(ctx context.Context, client *cvm.Client, imageName string) (
 }
 
 // NewCvmClient returns a new cvm client
-func NewCvmClient(secretId, secretKey, region string) (client *cvm.Client, err error) {
+func NewCvmClient(secretId, secretKey, region, endpoint string) (client *cvm.Client, err error) {
 	cpf := profile.NewClientProfile()
 	cpf.HttpProfile.ReqMethod = "POST"
 	cpf.HttpProfile.ReqTimeout = 300
 	cpf.Language = "en-US"
+	if endpoint != "" {
+		u, _ := url.Parse(endpoint)
```

Comment:
We should probably catch the error here, if the URL is malformed we can return an error to the caller.